cpp/src/parquet/metadata.h

```diff
@@ -24,10 +24,15 @@
 #include <vector>
 
 #include "arrow/util/key_value_metadata.h"
-
+#include <boost/any.hpp>
```

Comment:
Please don't add boost includes to any public headers

Reply:
I am not using boost, I think the final commit doesn't have that header